src/Microsoft.Restier.WebApi/Query/QueryTotalCount.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Restier.Core.Query;
+
+namespace Microsoft.Restier.WebApi.Query
+{
+    internal class QueryTotalCount : IQueryExecutor
+    {
```

Comment:
Yeah, I think your suggestion is good given that we may support multiple protocols in the future. But I'd suggest we keep consistency on the naming for now (at least before GA) unless we decide to add `OData` to all `RestierXXX` classes. When the time comes that we need to implement a new publisher for OneAPI, we may have more considerations. Some classes (e.g., serializers and deserializers) are not shared by OData and OneAPI then we can give them different names as you suggested. But for some shared classes (e.g., this query executor because OneAPI also has count option), we may give them the same names. Anyway we need to figure out a distinguishable and consistent set of names at that time.